src/app/[id]/page.tsx

```diff
@@ -0,0 +1,160 @@
+        <Typography variant="body1" color="error">
+          Failed to load product details.
+        </Typography>
+      </Box>
+    );
+  }
+
+  const discountPrice = product.discountPercentage
+    ? calculateDiscountedPrice(product.price, product.discountPercentage)
+    : undefined;
+
+  return (
+    <ThemeWrapper>
+      <MainSection {...product} discountPrice={discountPrice} />
+    </ThemeWrapper>
+  );
+};
+
+export const generateStaticParams = async () => {
+  try {
+    const response = await fetch("https://dummyjson.com/products");
+    const data = await response.json();
+
+    const products = data.products;
+
+    if (!Array.isArray(products)) {
+      console.error("Expected an array of products but got:", products);
+      return [];
+    }
+
+    return products.map((product: ProductType) => ({
+      id: product.id.toString(),
+    }));
+  } catch (error) {
+    console.error("Failed to fetch products:", error);
+
+    return [];
+  }
+};
+
+const MainSection = ({
+  discountPrice,
+  ...data
+}: ProductType & { discountPrice?: number }) => {
+  return (
+    <>
+      <Box component="main" sx={styles.main}>
+        <Box sx={styles.imageBox}>
+          {data.images.length === 1 ? (
+            <Box
+              component="img"
+              src={data.images[0]}
+              alt={"Image"}
+              loading="lazy"
+              sx={styles.image}
+            />
+          ) : (
+            <ImageCarousel images={data.images} />
+          )}
+        </Box>
+
+        <Box sx={styles.contentBox}>
+          <Typography variant="h4">{data.title}</Typography>
+
+          <Box sx={styles.ratingBox}>
+            <Typography variant="body1">
+              {data.rating && `${data.rating}`}
+            </Typography>
+            <Rating
+              ratingInPercent={data.rating}
+              iconSize="l"
+              showOutOf={true}
+            />
+            <Typography variant="body1">{`(${data.reviews.length} reviews)`}</Typography>
+          </Box>
+
+          <Box sx={styles.priceAndCartBox}>
+            <Box>
+              <Typography
+                variant="body2"
+                color="text.secondary"
+                sx={
+                  data.discountPercentage
+                    ? { textDecoration: "line-through" }
+                    : undefined
+                }
+              >
+                {`$${data.price}`}
+              </Typography>
+              {data.discountPercentage && (
+                <Typography variant="h5" color="red">
+                  {`$${discountPrice}`}
+                </Typography>
+              )}
+            </Box>
+            <AddToCart data={data} />
+          </Box>
+
+          <Typography
+            variant="body1"
+            sx={{
+              ...styles.stockStatus,
+              ...(data.availabilityStatus === "Low Stock"
+                ? styles.redText
+                : styles.greenText),
+            }}
```

Comment:
Oh, that is hard to read, consider adding some util function like `cn` but for [mui sx](https://www.npmjs.com/package/merge-sx)